jib-maven-plugin/src/main/java/com/google/cloud/tools/jib/maven/MavenSettingsServerCredentials.java

```diff
@@ -108,4 +109,23 @@ public String getPasswordDescriptor() {
           }
         });
   }
+
+  @Nullable
+  Server getServerFromMavenSettings(String registry) {
+    // Find and remove port (if present)
+    if (registry == null) {
+      return null;
+    }
+
+    Server server = settings.getServer(registry);
+    if (server != null) {
+      return server;
+    }
+
+    int index = registry.lastIndexOf(':');
+    if (index != -1) {
+      registry = registry.substring(0, index);
+    }
+    return settings.getServer(registry);
```

Comment:
I suggest

```java
    int index = registry.lastIndexOf(':');
    if (index != -1) {
      return settings.getServer(registry.substring(0, index));
    }
    return null;
```
No need to call `settings.getServer()` twice if nothing is configured.